src/nvim/syntax.c

```diff
@@ -6827,6 +6827,9 @@ int hl_combine_attr(int char_attr, int prim_attr)
     new_en = *char_aep;
   } else {
     memset(&new_en, 0, sizeof(new_en));
+    new_en.rgb_fg_color = -1;
```

Comment:
Alternatively, maybe this entire branch can be removed? `new_en` already has the correct `ATTRENTRY_INIT` value before the `memset()`.